core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -2102,6 +2111,10 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
                 read_leb_uint32(frame_ip, frame_ip_end, reserved);
                 delta = (uint32)POP_I32();
 
+#if WASM_ENABLE_SHARED_MEMORY != 0
+                if (node)
+                    os_mutex_lock(&node->shared_mem_lock);
```

Comment:
Should we also add the lock for aot_enlarge_memory:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/aot/aot_runtime.c#L1833

Reply:
Or maybe I can move the lock in `wasm_enlarge_memory()` directly